---
title: "ABSTRACT (ESA 2017): Genomics of the Genus
Aphaenogaster Reveals Biogeographic Variation in Key Ecosystem and
Climate Change Traits"
author: M.K. Lau, A.M. Ellison, B. DiMarco, A.Nguyen, S.H. Cahan,
N.J. Sanders, R. Dunn, C. Pennick, N.J. Gotelli
output: html_document
bibliography: mendeley.bib
---


### Abstract

- Climate change rates are currently on track with the worst case
  sencarios, resulting in both terrestrial and aquatic ecosystem responses
- Ants play important roles in ecosystems, including seed dispersal,
  which is a know factor in determining the rates of shifts in the
  range of many plant species
- Here, we present an analysis of a set of genomic sequences from six
  species in the genus Aphaenogaster, which is ecologically important
  in eastern forest ecosystems and has been shown to respond to
  changes in atmospheric temperature

- Analysis: Determine target analyses from recent ant and climate
  change genomics papers

- Results: 
- Conclusion:
- Take-Home:

### Methods 

The complete genome of six species of Aphaenogaster were sequenced:
Aphaenogaster ashmeadi, A. floridana, A. fulva, A. miamiana, A. picea
and A. rudis. Whole colonies were collected from field sites in
Eastern North America. Living individuals from each colony were
isolated from debris, placed in 50 ml Falcon centrifuge tubes and
immediately flash frozen in a -80 C freezer. DNA was extracted from
each colony and sequenced on an Illumina HiSeq 2500 at the Broad
Institute (Cambridge, MA). Raw sequences were processed to remove
chimeric and contaminant sequences and assembled using ALLPATHS-LG
with an additional post-assembly processing using Pilon to reduce gaps
in coverage. The genomes of all six species were successfully
sequenced with an average genome size of ~470 Mb. All genomes showed
high assembly quality (70% fragments mapped). The additional Pilon
processing of the assembly reduced the gaps in coverage by 3.1% on
average for an average reduction of 3.9 Mb of gaps. Both raw and
assembly data are currently housed at Harvard University on both a
long-term storage server that is routinely backed-up and a solid state
external hard drive stored at Harvard Forest (Petersham,
MA). Additionally, after publication the assemblies will be submitted
to NCBIâ€™s genome database and linked to meta-data hosted by the
Harvard Forest Data Archive.

### Sequencing Details

0. Sample locations
1. Genetics of ants and colonies
2. DNA extraction and quality
3. Library perparation
4. Genome assembly de novo (draft genome)
5. Assembly quality assessment and validation
6. Genome annotation (GO, KEGG, protein coding sequences)


### Genome Annotation

0. Gather target genes from other ant, insect and animal research
1. Mask repeat regions
2. Map target sequences
3. Compare translated sequences

Check out these software packages:

MAKER
PASA
WebApollo
GMOD



### Analysis Outline

Basic analysis of genomic similarity across species
Basic analysis of target genes

1. Heat stress response genes @Stanton-Geddes2016
2. Cold tolerance genes @Stanton-Geddes2016
3. Map transcripts onto genome and look for inter-specific variance @Stanton-Geddes2016
4. Immunity @Roux2014
5. Olfaction @Roux2014
6. Mitochondrial function @Roux2014
7. Social chromosome? @Brelsford2014
